cmd/node-cache/app/cache_app_test.go

```diff
@@ -186,3 +186,79 @@ func TestUpdateCoreFile(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func TestUpdateIPv6CoreFile(t *testing.T) {
+	baseDir, err := ioutil.TempDir("", "dnstest")
+	if err != nil {
+		t.Fatalf("Failed to obtain temp directory for testing, err %v", err)
+	}
+	envName := strings.ToUpper(strings.Replace(UpstreamClusterDNS, "-", "_", -1)) + "_SERVICE_HOST"
+	os.Setenv(envName, "2001:db8::1")
+	defer func() { os.RemoveAll(baseDir) }()
+	c, err := NewCacheApp(&ConfigParams{LocalIPStr: "fe80:169:254::1,fd00:1:2:3::5",
```

Comment:
Is "fd00:1:2:3::5" the v6 kube-dns IP and "fe80:169:254::1" the 169.254.20.10 counterpart? We should document this IP for v6 clusters, in https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/#configuration

Reply:
My reading is that these are configuration options, right? An user not necessarily have to use those values, or do they?
I can update the docs using IPv6 addresses, but want to clarify that in IPv4 the link-local addresses are reserved to 169.254.0.0/16. For IPv6 the reserved range is fe80::/10, however, unlike IPv4, IPv6 requires a link-local address on every network interface on which the IPv6 protocol is enabled,.

IIRC 10.0.0.10 is the service ClusterIP, if that is right that will depend on the ServiceSubnet configured

Reply:
I will submit a follow up PR to update the doc to mention you can use IPv6 addresses and adding some examples

Reply:
> My reading is that these are configuration options, right? An user not necessarily have to use those values, or do they?
True, the user can use any value for the listen ip. We do provide one example value for V4 - 169.254.20.10. It would be good to list one such example value for V6 and include a note that any linklocal IP is ok.

Reply:
https://github.com/kubernetes/website/pull/24957

Reply:
> IPv6 requires a link-local address on every network interface on which the IPv6 protocol is enabled

I'm trying to imagine what netns the node-local DNS could run in to make this work. Link-local traffic doesn't route, so does the node-local DNS run in the netns of the Pod? If so, which one? Maybe the node-local DNS runs in its own netns and something does route the link-local traffic to it?

It's hard to imagine - maybe there is a diagram that helps?

Reply:
BTW I hope that 169.254.20.10 is a _documentation example_ and not a default. Unless you _know_ you control the link and the order that things arrive onto it, it's wrong to assume that you can pick any IPv4 address inside `169.254.0.0/16` and have that available. Given the diverse set of container network plugins, I don't think that assumption is safe.

Reply:
it' s a doc example, you can use whatever IP, it seems that the pod creates an interface with that IP inside the node, but there is nothing special about it

Reply:
yes, it can be any IP as long as it is guaranteed to not collide with anything else on the node. Nothing else special about it.